python/NOTICE

```diff
@@ -6,3 +6,22 @@ This product includes software developed at
 The Apache Software Foundation (http://www.apache.org/).
 
 --------------------------------------------------------------------------------
+
+This project includes code from Apache Avro from the Apache Software Foundation with
+the following copyright notice:
```

Comment:
We don't need to add copyright notices into NOTICE. This only needs to include relevant NOTICE contents from other projects. For Avro, there's nothing that applies to the code that we're including because all of the NOTICE contents are for the C# implementation. So you can remove this.

We do need to update the LICENSE file for Python to cover the parts of Avro that we copied. Here's how we did that for the Java project: https://github.com/apache/iceberg/blob/master/LICENSE#L214-L223